### PR TITLE
Enhance sentinel_linear_search doctests with comprehensive examples

### PR DESCRIPTION
**Pull request by JesunAhmadUshno**
## Summary
Enhanced doctest coverage for `sentinel_linear_search()` with 15 comprehensive test examples covering:
- Valid searches (first, middle, last element)
- Invalid searches (element not found)
- Edge cases (empty list, single-element list)
- Type variations (integers, strings, floats, negative numbers)
- Duplicates (finding first occurrence of duplicated values)

## Changes
- Added 11 new doctest examples (from 4 → 15 total)
- Improved docstring documentation explaining the algorithm's sentinel approach
- All tests pass locally: `python -m doctest -v sentinel_linear_search.py`

## Checklist
- [x] I have read the CONTRIBUTING.md file
- [x] This is my own work and not plagiarized
- [x] My work will be distributed under the MIT License once merged
- [x] My submission follows the coding style and standards of this repository
- [x] I have tested my changes locally
- [x] All doctest examples pass
- [x] Code passes ruff linting

### CODE REVIEW

File: searches/sentinel_linear_search.py [hunk 1]
```diff
@@ -18,6 +18,11 @@ def sentinel_linear_search(sequence, target):
     :param target: item value to search
     :return: index of found item or None if item is not found
 
+    The sentinel linear search algorithm works by appending the target value to the
+    end of the list before searching. This eliminates the need for boundary checking
+    during the loop, as the search is guaranteed to find the target (at minimum, at
+    the end where the sentinel is placed).
```
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply searches/sentinel_linear_search.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343768)

File: graphs/check_bipatrite.py [hunk 2]
```diff
@@ -113,7 +113,6 @@ def is_bipartite_bfs(graph: dict[int, list[int]]) -> bool:
     >>> is_bipartite_bfs({7: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 4: [0]})
     False
 
-    >>> # FIXME: This test should fails with KeyError: 4.
     >>> is_bipartite_bfs({0: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 9: [0]})
     False
```
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply graphs/check_bipatrite.py:117]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343737)

File: graphs/check_bipatrite.py [hunk 3]
```diff
@@ -33,7 +35,6 @@ def is_bipartite_dfs(graph: dict[int, list[int]]) -> bool:
     >>> is_bipartite_dfs({7: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 4: [0]})
     False
 
-    >>> # FIXME: This test should fails with KeyError: 4.
     >>> is_bipartite_dfs({0: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 9: [0]})
     False
```
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply graphs/check_bipatrite.py:39]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343718)

File: searches/jump_search.py [hunk 4]
```diff
@@ -10,17 +10,14 @@
...
 class Comparable(Protocol):
     def __lt__(self, other: Any, /) -> bool: ...
 
 
-T = TypeVar("T", bound=Comparable)
-
-
-def jump_search(arr: Sequence[T], item: T) -> int:
+def jump_search[T: Comparable](arr: Sequence[T], item: T) -> int:
     """
```
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply searches/jump_search.py:21]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343673)

[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply searches/jump_search.py:21]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343673)

File: graphs/check_bipatrite.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply graphs/check_bipatrite.py:39]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343718)

File: graphs/check_bipatrite.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply graphs/check_bipatrite.py:117]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343737)

File: searches/sentinel_linear_search.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JesunAhmadUshno) [reply searches/sentinel_linear_search.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/TheAlgorithms/Python/pull/14284#discussion_r2816343768)

**Review by mindaugl — APPROVED** [review]

**Comment by cclauss** [comment]
Multiple algorithms are modified.
